solidity/contracts/utility/TokenHolder.sol

```diff
@@ -13,32 +13,66 @@ import "./interfaces/ITokenHolder.sol";
  *
  * We consider every contract to be a 'token holder' since it's currently not possible
  * for a contract to deny receiving tokens.
- *
- * Note that we use the non standard ERC-20 interface which has no return value for transfer
- * in order to support both non standard as well as standard token contracts.
- * see https://github.com/ethereum/solidity/issues/4116
  */
 contract TokenHolder is ITokenHolder, Owned, Utils {
     using SafeERC20 for IERC20;
 
+    // prettier-ignore
+    receive() external payable override virtual {}
+
     /**
-     * @dev withdraws tokens held by the contract and sends them to an account
+     * @dev withdraws funds held by the contract and sends them to an account
      * can only be called by the owner
      *
-     * @param _token   ERC20 token contract address
-     * @param _to      account to receive the new amount
-     * @param _amount  amount to withdraw
+     * @param token ERC20 token contract address (with a special handling of NATIVE_TOKEN_ADDRESS)
+     * @param to account to receive the new amount
+     * @param amount amount to withdraw
      */
     function withdrawTokens(
-        IERC20 _token,
-        address _to,
-        uint256 _amount
-    ) public virtual override ownerOnly validAddress(address(_token)) validAddress(_to) notThis(_to) {
-        if (_token == NATIVE_TOKEN_ADDRESS) {
-            payable(_to).transfer(_amount);
+        IERC20 token,
+        address to,
+        uint256 amount
+    ) public virtual override ownerOnly validAddress(to) {
```

Comment:
can change to `validExternalAddress` once we merge the other PR

Reply:
Are you sure you want `validExternalAddress` instead of `validAddress`? In the previous PR, we have explicitly removed the need for the `notThis` check in this case.

Reply:
not critical